quarry/web/api.py

```diff
@@ -150,47 +150,6 @@ def api_run_query() -> Tuple[Union[str, Response], int]:
     )
 
 
-@api_blueprint.route("/api/query/stop", methods=["POST"])
-def api_stop_query() -> Tuple[Union[str, Response], int]:
-    if get_user() is None:
-        return "Authentication required", 401
-
-    qrun_id = request.form["qrun_id"]
-    db_of_process = request.form["query_database"]
-
-    # the db process id of the running job is stored in the query_run table while
-    # the job is running. We can take this pid over to the database running the
-    # query to stop the job
-    query_run = (
-        g.conn.session.query(QueryRun).filter(QueryRun.id == qrun_id).one()
-    )
-    result_dictionary = ast.literal_eval(query_run.extra_info)
-    if "connection_id" in result_dictionary:
-        g.replica.connection = db_of_process
```

Comment:
fwiw, this seems obviously wrong - `db_of_process` is a parameter with the database name, so it doesn't have a `.cursor()` method that the next line uses

Reply:
it's fine. `.connection =` on this line invokes the setter in the Replica class. On the next line, it's a getter which gives the underlying pymysql connection which does have a `.cursor()` method.